File: ARCADE/Intro/Level 11_2: lineEncoding.cs
    using System;
     
    public class Test
    {
    	public static string lineEncoding(string s) {
    	    string s1 = "";
    	    int count = 1, n = 0;
    	    for(int i = 1; i < s.Length; i++){
    	        n = i;
    	        if(s[i] == s[i - 1]) count++;
    	        else{
    	            if(count == 1) s1 += s[i-1];
    	            else {
    	                s1 += count.ToString() + s[i-1];
    	                count = 1;
    	            }
    	        }
    	    }
     
    	    if(count == 1) s1 += s[n];
    	    else {
    	        s1 += count.ToString() + s[n];
    	        count = 1;
    	    }
    	    return s1;
    	}
     
    	public static void Main()
    	{
    		string s = "aabbbc";
    		Console.Write(lineEncoding(s));
    	}
    }

File: ARCADE/Intro/Level 11_2: lineEncoding.pas
program lineEncoding;
uses Sysutils;

function lineEncoding(s: string): string;
var i, j, count : integer;
    rs : string;
begin
    i := 1; rs := '';
    while(i <= length(s)) do
    begin
        count := 1;
        j := i;
        while(j < length(s)) do
        begin
            j := j + 1;
            if(s[i] = s[j]) then inc(count)
            else break;
        end;
        if(count <> 1) then rs := rs + IntToStr(count) + s[i]
        else rs := rs + s[i];
        inc(i, count);
    end;
    
    lineEncoding := rs;
end;

var s : string;
begin
	s := 'aabbbc';
	writeln(lineEncoding(s));
end.
